metaserver/src/main/scala/scala/meta/languageserver/Compiler.scala

```diff
@@ -1,54 +1,52 @@
 package scala.meta.languageserver
 
 import java.io.File
+import java.io.PrintStream
 import java.nio.file.Files
 import java.util.Properties
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentSkipListSet
 import scala.collection.mutable
+import scala.concurrent.Future
 import scala.reflect.io
+import scala.runtime.CharRef
 import scala.tools.nsc.Settings
 import scala.tools.nsc.interactive.{Global, Response}
 import scala.tools.nsc.reporters.StoreReporter
 import com.typesafe.scalalogging.LazyLogging
 import langserver.core.Connection
 import langserver.messages.MessageType
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.reactive.MulticastStrategy
 import monix.reactive.Observable
 import org.langmeta.io.AbsolutePath
+import org.langmeta.semanticdb.Document
 
-case class CompilerConfig(
-    sources: List[AbsolutePath],
-    scalacOptions: List[String],
-    classpath: String
-)
-object CompilerConfig {
-  def fromPath(
-      path: AbsolutePath
-  )(implicit cwd: AbsolutePath): CompilerConfig = {
-    val input = Files.newInputStream(path.toNIO)
-    try {
-      val props = new Properties()
-      props.load(input)
-      val sources = props
-        .getProperty("sources")
-        .split(File.pathSeparator)
-        .iterator
-        .map(AbsolutePath(_))
-        .toList
-      val scalacOptions = props.getProperty("scalacOptions").split(" ").toList
-      val classpath = props.getProperty("classpath")
-      CompilerConfig(sources, scalacOptions, classpath)
-    } finally input.close()
-  }
-}
 class Compiler(
+    out: PrintStream,
     config: Observable[AbsolutePath],
     connection: Connection,
     buffers: Buffers
-)(implicit cwd: AbsolutePath)
+)(implicit cwd: AbsolutePath, s: Scheduler)
     extends LazyLogging {
+  private val documentPubSub =
```

Comment:
This name makes me think it is (Publisher, Subscriber) but based on the usage it is actually (Subscriber, Publisher)

Reply:
Fixed.